### DISCUSSION TITLE
Should `biome format`/`biome lint` run the formatter/linter even if they are disabled?

### DISCUSSION THREAD

**Discussion by Conaclos** [biomejs/biome#2865] [General]
While I was working on #2860, I noticed that `biome format` and `biome lint` don't run the formatter / the linter when they are disabled in the configuration.
For instance, given the following configuration:

```json5
{
  "formatter": { "enabled": false },
  "linter": { "enabled": false },
  "overrides": [{
    "include": ["dist/**/*"],
    "formatter": { "enabled": false },
    "linter": { "enabled": false },
  }]
}
```

The following commands don't execute the formatter and the linter.

```shell
❯ npx biome lint .
❯ npx biome format .
```

Unlike these commands, `biome check` allows overriding  `linter>enabled` and `formatter.enabled` using the corresponding CLI option.
With the same configuration, a user can force the execution of the formatter and the linter:

```shell
❯ npx biome check --formatter-enabled=true --linter-enabled=true .
```

Note that this doesn't override `overrides`. Thus, the formatter and the linter are still disabled for files under the dist directory.

I don't think we should add the `--formatter-enabled` and `--linter-enabled` on the `biome format` and the `biome lint` commands. Instead, we could consider that they are implied.
If a user execute `biome format`/`biome lint`, I think it is fair to assume that the user wants to enable the formatter/the linter.

**Comment by ematipico** [comment]
I don't understand the proposal, sorry. 

`biome format` doesn't accept `--formatter-enabled`, and `biome lint` doesn't accept `--linter-enabled`.

**Reply by Conaclos** [reply]
The proposal is to assume that `linter.enabled` and `formatter.enabled` are `true` when running `biome lint`/`biome format`.

**Reply by ematipico** [reply]
Usually, I'm not very fond of "magic", and this feels like it, because the user doesn't see it.

I would rather throw an error in case the user tries to run the formatter but it's disabled via configuration.

**Comment by arendjr** [comment]
I think I might lean towards a simple `--force` option that will allow running these commands even if disabled? That is also easy to explain in the output, something like:

```
$ biome format
The formatter is disabled in **insert config file path**. Please update your configuration or run `biome format --force` to format anyway.
```

It's *probably* disabled for a reason in the config, so assuming it should be overridden might be a bit of a leap. But it is true the user might just want to experiment or format/lint something specifically, and it's annoying if you are required to update the config for that.
